Source/ASImageNode+AnimatedImage.mm

```diff
@@ -70,13 +70,23 @@ - (void)_locked_setAnimatedImage:(id <ASAnimatedImageProtocol>)animatedImage
       [self _locked_setShouldAnimate:YES];
     }
   } else {
-      // Clean up after ourselves.
-      self.contents = nil;
-      [self setCoverImage:nil];
-  }
-  
-  [self animatedImageSet:_animatedImage previousAnimatedImage:previousAnimatedImage];
+    // Clean up after ourselves.
     
+    // Don't bother using a `_locked` version for setting contnst as it should be pretty safe calling it with
+    // reaquire the lock and would add overhead to introduce this version
+    self.contents = nil;
+    [self _locked_setCoverImage:nil];
+  }
+
+  // Push calling subclass to the next runloop cycle
+  // We have to schedule the block on the common modes otherwise the tracking mode will not be included and it will
+  // not fire e.g. while scrolling down
+  CFRunLoopPerformBlock(CFRunLoopGetCurrent(), kCFRunLoopCommonModes, ^(void) {
+    [self animatedImageSet:animatedImage previousAnimatedImage:previousAnimatedImage];
+  });
+  // Don't need to wakeup the runloop as the current is already running
+// CFRunLoopWakeUp(runLoop); // Should not be necessary
+
   // Animated image can take while to dealloc, do it off the main queue
   if (previousAnimatedImage != nil) {
```

Comment:
I think we need to deallocate the previous image after calling out to subclass, otherwise it will not be deallocated in background if the deallocation occurs before the next run loop.